test/test-fs.c

```diff
@@ -4533,3 +4533,53 @@ TEST_IMPL(fs_get_system_error) {
 
   return 0;
 }
+
+#ifdef _WIN32
+TEST_IMPL(fs_wtf) {
+  int r;
+  HANDLE file_handle;
+  uv_dirent_t dent;
+  static char test_file_buf[PATHMAX];
+
+  /* set-up */
+  _wunlink(L"test_dir/hi\xD801\x0037");
+  rmdir("test_dir");
+
+  loop = uv_default_loop();
+
+  uv_fs_mkdir(NULL, &mkdir_req, "test_dir", 0777, NULL);
+  uv_fs_req_cleanup(&mkdir_req);
+
+  file_handle = CreateFileW(L"test_dir/hi\xD801\x0037",
+                            GENERIC_WRITE | FILE_WRITE_ATTRIBUTES,
+                            0,
+                            NULL,
+                            CREATE_ALWAYS,
+                            FILE_FLAG_OPEN_REPARSE_POINT |
+                              FILE_FLAG_BACKUP_SEMANTICS,
+                            NULL);
+  ASSERT(file_handle != INVALID_HANDLE_VALUE);
+
+  CloseHandle(file_handle);
+
+  r = uv_fs_scandir(NULL, &scandir_req, "test_dir", 0, NULL);
+  ASSERT_EQ(r, 1);
+  ASSERT_EQ(scandir_req.result, 1);
+  ASSERT_NOT_NULL(scandir_req.ptr);
+  while (UV_EOF != uv_fs_scandir_next(&scandir_req, &dent)) {
+    strcpy(test_file_buf, "test_dir\\");
+    strcat(test_file_buf, dent.name);
+    r = uv_fs_stat(NULL, &stat_req, test_file_buf, NULL);
```

Comment:
This is failing because we (correctly) read `L"hi\xd801\067"` and it is (correctly) converted to:
```
(gdb) p/x wtf8[0]
$8 = 0x68
$9 = 0x69
$10 = 0xed
$11 = 0xa0
$12 = 0x81
$13 = 0x37
$14 = 0x0
```

Then when we try to convert this back, the `fs__decode_wtf8_char` function fails near the last character because the state machine there gets to the following state 
```
(gdb) info loc
code_point = 972801
b1 = 237 '\355'
b2 = 160 '\240'
b3 = 129 '\201'
b4 = 55 '7'
```
and then deems
```
176       if ((b4 & 0xc0) != 0x80)
177         return -1; /* invalid: not a continuation byte */
```